examples/Phonon_Caustics/src/Caustic_PhononSensitivity.cc

```diff
@@ -0,0 +1,96 @@
+/***********************************************************************\
+ * This software is licensed under the terms of the GNU General Public *
+ * License version 3 or later. See G4CMP/LICENSE for the full license. *
+\***********************************************************************/
+
+//20240110 Israel Hernandez -- Illinois Institute of Technology
+#include "Caustic_PhononSensitivity.hh"
+#include "G4CMPElectrodeHit.hh"
+#include "G4Event.hh"
+#include "G4HCofThisEvent.hh"
+#include "G4PhononLong.hh"
+#include "G4PhononTransFast.hh"
+#include "G4PhononTransSlow.hh"
+#include "G4Run.hh"
+#include "G4RunManager.hh"
+#include "G4SDManager.hh"
+#include "G4SystemOfUnits.hh"
+#include "Caustic_PhononConfigManager.hh"
+#include <fstream>
+//20240110 Israel Hernandez -- Illinois Institute of Technology
+
+Caustic_PhononSensitivity::Caustic_PhononSensitivity(G4String name) :
+  G4CMPElectrodeSensitivity(name), fileName("") {
+  SetOutputFile(Caustic_PhononConfigManager::GetHitOutput());
+}
+
+
+
+Caustic_PhononSensitivity::~Caustic_PhononSensitivity() {
+  if (output.is_open()) output.close();
+  if (!output.good()) {
+    G4cerr << "Error closing output file, " << fileName << ".\n"
+           << "Expect bad things like loss of data.";
+  }
+}
+
+void Caustic_PhononSensitivity::EndOfEvent(G4HCofThisEvent* HCE) {
+  G4int HCID = G4SDManager::GetSDMpointer()->GetCollectionID(hitsCollection);
+  auto* hitCol = static_cast<G4CMPElectrodeHitsCollection*>(HCE->GetHC(HCID));
+  std::vector<G4CMPElectrodeHit*>* hitVec = hitCol->GetVector();
+
+  G4RunManager* runMan = G4RunManager::GetRunManager();
+
+  if (output.good()) {
+    //Saving in a txt file the Final Phonon Position.
+    for (G4CMPElectrodeHit* hit : *hitVec) {
+        output << hit->GetParticleName() << '\t'
+        << hit->GetFinalPosition().getX()/m << '\t'
+        << hit->GetFinalPosition().getY()/m << '\t'
+        << hit->GetFinalPosition().getZ()/m << '\n';
+
+    }
+  }
+}
```

Comment:
Just as a note, this is not thread-safe.  You're going to end up either having each thread open and write to the same file, or you'll have a file opened by master and all the threads writing to it.  Either way, the result will be a mish-mash of overlapping data.  

Making this thread safe could be done straightforwardly by using G4Analaysis, defining a tuple, and having G4Analysis write out a CSV (or tab-delimited) file.